File: .agents/skills/skills-add/scripts/register-skill-lock.py
#!/usr/bin/env python3
"""register-skill-lock — スキルを skills.lock.json の external セクションに登録する

使い方:
  python3 lib/register-skill-lock.py LOCK_FILE SKILL_NAME SOURCE SOURCE_URL SKILL_PATH

例:
  python3 lib/register-skill-lock.py skills.lock.json copywriting coreyhaines31/marketingskills \
    https://github.com/coreyhaines31/marketingskills.git skills/copywriting/SKILL.md
"""

import json
import os
import sys


def main():
    lock_path = sys.argv[1]
    skill_name = sys.argv[2]
    source = sys.argv[3]
    source_url = sys.argv[4]
    skill_path = sys.argv[5]

    with open(lock_path) as f:
        lock = json.load(f)

    if "external" not in lock:
        lock["external"] = {}

    lock["external"][skill_name] = {
        "source": source,
        "sourceUrl": source_url,
        "skillPath": skill_path,
    }

    tmp = lock_path + ".tmp"
    with open(tmp, "w") as f:
        json.dump(lock, f, indent=2)
        f.write("\n")
    os.replace(tmp, lock_path)
    print("  -> Registered in skills.lock.json")


if __name__ == "__main__":
    main()

File: .agents/skills/skills-add/scripts/register-skill-lock.ts
#!/usr/bin/env bun
/**
 * register-skill-lock — スキルを skills.lock.json の external セクションに登録する（TypeScript 版）
 *
 * Python 版 register-skill-lock.py を挙動互換で置き換える。
 * コマンドライン契約（args・stdout・JSON 変異・アトミック書き込み）は不変。
 *
 * 使い方:
 *   bun register-skill-lock.ts LOCK_FILE SKILL_NAME SOURCE SOURCE_URL SKILL_PATH
 */

import fs from "node:fs"
import process from "node:process"

type LockFile = {
  external?: Record<string, { source: string; sourceUrl: string; skillPath: string }>
  [key: string]: unknown
}

function main(): void {
  const [lockPath, skillName, source, sourceUrl, skillPath] = process.argv.slice(2)
  if (!lockPath || !skillName || !source || !sourceUrl || !skillPath) {
    console.error("Error: register-skill-lock requires LOCK_FILE SKILL_NAME SOURCE SOURCE_URL SKILL_PATH")
    process.exit(2)
  }

  let lock: LockFile
  try {
    lock = JSON.parse(fs.readFileSync(lockPath, "utf-8")) as LockFile
  } catch {
    console.error(`Error: Cannot read or parse lock file: ${lockPath}`)
    process.exit(1)
  }

  if (typeof lock !== "object" || lock === null || Array.isArray(lock)) {
    console.error(`Error: Lock file is not a JSON object: ${lockPath}`)
    process.exit(1)
  }

  if (typeof lock.external !== "object" || lock.external === null) {
    lock.external = {}
  }

  lock.external[skillName] = { source, sourceUrl, skillPath }

  const tmp = `${lockPath}.tmp`
  fs.writeFileSync(tmp, `${JSON.stringify(lock, null, 2)}\n`)
  fs.renameSync(tmp, lockPath)
  process.stdout.write("  -> Registered in skills.lock.json\n")
}

if (import.meta.main) {
  main()
}